src/CMakeLists.txt

```diff
@@ -871,16 +871,14 @@ add_subdirectory(compressor)
 
 add_subdirectory(tools)
 
-# dmClock
-
-add_subdirectory(dmclock) # after gmock
-add_dependencies(tests dmclock-tests dmclock-data-struct-tests)
+# dmClock (after gmock)
 
+add_subdirectory(dmclock/src)
 if(WITH_TESTS)
-  install(PROGRAMS
-    ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/dmclock-tests
-    ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/dmclock-data-struct-tests
-    DESTINATION bin)
+  # note: add_test is not being called, so dmclock tests aren't part
+  # of ceph tests
+  add_subdirectory(dmclock/test)
```

Comment:
This is a polite request, by the way :-) I'm happy to continue discussing it. I'm also preparing a PR against master to demonstrate what I mean.